### ISSUE TITLE
Parser should allow lambdas with `ref` return types in assignments without requiring parentheses

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#54258]
The initializers for `d1` and `d2` should be equivalent. Currently, the initializer for `d2` is interpreted as `ref (int () => x)` and is reported as an error.
```csharp
delegate ref int D();

class Program
{
    static void Main()
    {
        D d1 = (ref int () => x);
        D d2 = ref int () => x;
    }
}
```
See [LDM notes](https://github.com/dotnet/csharplang/blob/main/meetings/2021/LDM-2021-06-21.md).

**Comment by jcouv** [comment]
FYI @RikkiGibson This is a candidate to offload from Chuck
